port/unix/omrsysinfo.c

```diff
@@ -510,14 +508,19 @@ omrsysinfo_get_OS_type(struct OMRPortLibrary *portLibrary)
 		rc = uname(&sysinfo);
 #endif
 		if (rc >= 0) {
-			len = strlen(sysinfo.sysname) + 1;
+			size_t len = 0;
+			char *buffer = NULL;
+			char *sysName = sysinfo.sysname;
+			if (0 == strcmp(sysName, "Darwin")) { /* Translate this to "OS X" */
```

Comment:
This should probably be ifdef'd so it only runs on osx.

Reply:
I could hard-code `sysName` and skip the call to `uname()`

Reply:
I think we should create a `#elif OSX` (using the proper define) and inside the code do something similar to this:
```
return "Mac OS X";
```

In the future if we have to change this based on the version of OS X or Mac OS running we already have a place to do it.